packages/jellyfish-api-core/__tests__/category/wallet/getBalances.test.ts

```diff
@@ -0,0 +1,42 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { BigNumber } from '../../../src'
+import { WalletFlag } from '../../../src/category/wallet'
+
+describe('getBalances on masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getBalances', async () => {
```

Comment:
We can add some tests here that returns the balance whenever we send some value on a wallet. Ex.
1. Send 1 DFI to an address
2. Call `getbalances`, check if the wallet balance is 1 DFI

Reply:
does not work like this, better to ask @canonbrother about this, he worked on the wallet system a long time ago too.

Reply:
I have added tests to show the balances after sending some amount of DFI out.

Reply:
actually you need to assert the behaviors of `mine, watchonly, trusted, untrusted_pending, immature, used`

Reply:
> actually you need to assert the behaviors of `mine, watchonly, trusted, untrusted_pending, immature, used`

@aikchun for a few behaviors you can ignore testing them, they required more setup that we don't currently have the capability to do yet. Do ping us if you don't know what they are.

One good example is this issue https://github.com/DeFiCh/jellyfish/issues/40, it's blocking may behavior that we can't test without having it.